lib/claim/token-holder-revenue-fund-contracts-ensemble.js

```diff
@@ -0,0 +1,359 @@
+                continue;
+
+            claimableFees = claimableFees.add(
+                await configuration.contract.claimableAmountByAccruals(
+                    wallet.address, currency.ct.toString(), currency.id,
+                    Math.max(startAccrual, span.startAccrual),
+                    Math.min(endAccrual, span.endAccrual)
+                )
+            );
+        }
+
+        return claimableFees;
+    }
+
+    /**
+     * Claim and stage for the given span of accruals
+     * @param {Wallet} wallet - The claimer nahmii wallet
+     * @param {Currency} currency - The currency
+     * @param {number} startAccrual - The lower accrual index boundary
+     * @param {number} endAccrual - The upper accrual index boundary
+     * @param [options] - An optional object containing the parameters for gasLimit and gasPrice
+     * @returns {Promise} A promise that resolves into an array of transaction hashes.
+     */
+    async claimAndStageByAccruals(wallet, currency, startAccrual, endAccrual, options) {
+        if (startAccrual > endAccrual)
+            throw new Error(`Ordinality mismatch of startAccrual > endAccrual (${startAccrual} > ${endAccrual})`);
+
+        if (!this.isSpanned(currency))
+            await this.span(currency);
+
+        const ensemble = _ensemble.get(this);
+        const currencyKey = currencyToKey(currency);
+
+        try {
+            const txs = [];
+
+            for (const configuration of ensemble) {
+                const span = configuration.spansByCurrency.get(currencyKey);
+
+                if (endAccrual < span.startAccrual || startAccrual > span.endAccrual)
+                    continue;
+
+                const contract = configuration.contract.connect(wallet);
+                txs.push(
+                    await contract.claimAndStageByAccruals(
+                        currency.ct.toString(), currency.id,
+                        Math.max(startAccrual, span.startAccrual),
+                        Math.min(endAccrual, span.endAccrual),
+                        options
+                    )
+                );
+            }
+
+            return txs;
+        }
+        catch (error) {
+            throw new NestedError(error, 'Unable to claim and stage by accruals.');
+        }
+    }
+
+    /**
+     * Get claimable amount for the given span of block numbers
+     * @param {Wallet} wallet - The claimer nahmii wallet
+     * @param {Currency} currency - The currency
+     * @param {number} startBlock - The lower block number boundary
+     * @param {number} endBlock - The upper block number boundary
+     * @returns {Promise} A promise that resolves into a BigNumber value representing the claimable amount.
+     */
+    async claimableAmountByBlockNumbers(wallet, currency, startBlock, endBlock) {
+        if (startBlock > endBlock)
+            throw new Error(`Ordinality mismatch of startBlock > endBlock (${startBlock} > ${endBlock})`);
+
+        if (!this.isSpanned(currency))
```

Comment:
Candidate for removal?

Reply:
Being replaced by `if (!isSpanned.call(this, currency))`.